File: Basics_with_Python/PythonBasicCourse/4_For_Loop/10_For_Loop_Exercise/More_Exercise2/01._Back_To_The_Past.py
inheritance = float(input())
year_to_live = int(input())
start_year = 1800
age = 17


for number in range(1800, year_to_live + 1):
    age +=1
    if number % 2 == 0:
        inheritance -= 12000
    else:
       inheritance -= 12000 + 50 * age

if inheritance >= 0:
    print(f"Yes! He will live a carefree life and will have {inheritance:.2f} dollars left.")
else:
    print(f"He will need {abs(inheritance):.2f} dollars to survive.")

File: Basics_with_JavaScript/910_For_Loop/More_Exersises/01._Back_To_The_Past.js
function vladi(array) {
  let budged = Number(array[0]);
  let endYear = Number(array[1]);
  let startYear = 1800;
  let startAge = 18;
  let currAge = startAge - 1;
  let diff = 0;

  for (let year = startYear; year <= endYear; year++) {
    currAge += 1;
    if (year % 2 === 0) {
      budged -= 12000;
    } else {
      budged = budged - 12000 - 50 * currAge;
    }
  }
  diff = Math.abs(budged).toFixed(2);
  if (budged >= 0) {
    console.log(
      `Yes! He will live a carefree life and will have ${diff} dollars left.`
    );
  } else {
    console.log(`He will need ${diff} dollars to survive.`);
  }
}

vladi(["50000", "1802"]);
vladi(["10000.15", "1808"]);
